compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2363,6 +2394,9 @@ pub struct TargetOptions {
     /// distributed with the target, the sanitizer should still appear in this list for the target.
     pub supported_sanitizers: SanitizerSet,
 
+    /// The stable sanitizers supported by this target
+    pub stable_sanitizers: SanitizerSet,
+
```

Comment:
If the policy is "Tier 1 ASAN and LeakSAN are stable", why are we adding this field here, instead of doing something like the following?
```rust
fn stable_sanitizers(&self) -> SanitizerSet {
    let mut sanitizers = SanitizerSet::empty();
    if self.metadata.tier == Some(1) {
        return sanitizers;
    }
    if self.supported_sanitizers.contains(SanitizerSet::ADDRESS) {
        sanitizers |= SanitizerSet::ADDRESS;
    }
    if self.supported_sanitizers.contains(SanitizerSet::LEAK) {
        sanitizers |= SanitizerSet::LEAK;
    }
    sanitizers
}
```

Reply:
We happen to be proposing the stabilisation of some sanitizers on tier 1 targets but we'll want to be able to stabilise sanitizers of other tier 2 targets, but maybe not all of them, and this approach gives us that flexibility.